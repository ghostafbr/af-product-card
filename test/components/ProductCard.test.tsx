import * as React from 'react';
import renderer, {ReactTestRenderer} from 'react-test-renderer';
import {ProductCard, ProductImage} from '../../src';
import {product1, product2} from '../data/products';

const {act} = renderer;

describe('ProductImage', () => {

    it('should display the component correctly', (): void => {
        const wrapper: ReactTestRenderer = renderer.create(
            <ProductCard product={product1}>
                {
                    () => (
                        <h1>Product Card</h1>
                    )
                }
            </ProductCard>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should increase the quantity of the product', (): void => {
        const value: number = 1;
        const wrapper: ReactTestRenderer = renderer.create(
            <ProductCard product={product1}>
                {
                    ({count, increaseBy}) => (
                        <>
                            <h1>Product Card</h1>
                            <span>{ count }</span>
                            <button onClick={ () => increaseBy(value) }>+</button>
                        </>
                    )
                }
            </ProductCard>
        );

        let tree = wrapper.toJSON();
        expect(tree).toMatchSnapshot();
        act(() => {
            (tree as any).children[2].props.onClick();
        });
        tree = wrapper.toJSON();
        expect((tree as any).children[1].children[0]).toBe('1');

    });

});
