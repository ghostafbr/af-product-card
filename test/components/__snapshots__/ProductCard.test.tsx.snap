// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should display the component correctly 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
</div>
`;

exports[`ProductImage should increase the quantity of the product 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
